server/activityItem/queries.ts

```diff
@@ -0,0 +1,340 @@
+	oldCommunity: types.Community,
+	communityId: string,
+) => {
+	const community: types.Community = await Community.findOne({ where: { id: communityId } });
+	const diffs = getDiffsForPayload(community, oldCommunity, ['title']);
+	return createActivityItem({
+		actorId,
+		kind,
+		communityId,
+		payload: {
+			...diffs,
+			community: {
+				title: community.title,
+			},
+		},
+	});
+};
+
+export const createMemberUpdatedActivityItem = async (
+	kind: 'member-updated',
+	actorId: string,
+	communityId: string,
+	memberId: string,
+	oldMember: types.Member,
+) => {
+	const member: types.Member = await Member.findOne({ where: { id: actorId } });
+	const diffs = getDiffsForPayload(member, oldMember, ['permissions']);
+	return createActivityItem({
+		kind,
+		actorId,
+		communityId,
+		payload: {
+			userId: memberId,
+			...diffs,
+		},
+	});
+};
+
+export const createMemberActivityItem = async (
+	kind: 'member-created' | 'member-removed',
+	actorId: string,
+	communityId: string,
+	memberId: string,
+) => {
+	const member: types.Member = await Member.findOne({ where: { id: actorId } });
+	return createActivityItem({
+		kind,
+		actorId,
+		communityId,
+		payload: {
+			userId: memberId,
+			permissions: member.permissions,
+		},
+	});
+};
+
+export const createCollectionActivityItem = async (
+	kind: 'collection-created' | 'collection-updated' | 'collection-removed',
+	collectionId: string,
+	actorId: string,
+	communityId: string,
+	oldCollection: types.Collection,
+) => {
+	const collection: types.Collection = await Collection.findOne({ where: { id: collectionId } });
+	const { title } = collection;
+	const diffs = getDiffsForPayload(collection, oldCollection, [
+		'isPublic',
+		'isRestricted',
+		'title',
+	]);
+	const flags = getChangeFlagsForPayload(collection, oldCollection, ['layout', 'metadata']);
+	return createActivityItem({
+		kind,
+		collectionId,
+		actorId,
+		communityId,
+		payload: {
+			collection: {
+				title,
+			},
+			...flags,
+			...diffs,
+		},
+	});
+};
+
+export const createPubReviewCreatedActivityItem = async (
+	kind: 'pub-review-created' | 'pub-review-comment-added',
+	actorId: string,
+	communityId: string,
+	reviewId: string,
+	isReply: boolean,
+) => {
+	const review: types.DefinitelyHas<types.Review, 'thread'> = await ReviewNew.findOne({
+		where: { id: reviewId },
+		includes: [
+			{ model: Thread, as: 'thread', includes: [{ model: ThreadComment, as: 'comments' }] },
+		],
+	});
+	const threadComment: types.ThreadComment = review.thread.comments[0];
+	const pub: types.Pub = await Pub.findOne({ where: { id: review.pubId } });
+	return createActivityItem({
+		communityId,
+		actorId,
+		kind,
+		pubId: pub.id,
+		payload: {
+			reviewId,
+			threadId: review.threadId,
+			isReply,
+			threadComment: {
+				id: threadComment.id,
+				text: threadComment.text,
+				userId: actorId,
+			},
+			pub: { title: pub.title },
+		},
+	});
+};
+
+export const createPubReviewUpdatedActivityItem = async (
```

Comment:
We don't need any `Thread`-related payload for this item (it's related strictly to changes in the `Review` model itself) so you can simplify this considerably!